src/codeu/chat/client/Controller.java

```diff
@@ -89,6 +90,31 @@ public User newUser(String name) {
     return response;
   }
 
+
+  public User restoreUser(User user) {
```

Comment:
Could you add javadoc comment what this does and what it is used for? I don't currently see when the client will be calling this method. There is similar method on the server (Model) that is called when we are repopulating users from the database, however I am not sure when this one is called on the client.